app/controllers/cards_controller.rb

```diff
@@ -49,6 +49,11 @@ def learning
     end
   end
 
+  def set_confidence_level
+    @card = @board.cards.find(params[:card_id])
+    redirect_to request.referer if @card.update(confidence_level: params[:confidence_level])
```

Comment:
it was said in the task: - `When you click on this button confidence_level updated. No redirection, just print "Mark saved" ( no need to print anything if it's difficult )`
so here we can jysy add flash notice & alert

Reply:
but if it requires too much js code, let's leave this variant